validator/sawtooth_validator/gossip/permission_verifier.py

```diff
@@ -28,46 +29,163 @@
+            LOGGER.debug("Batch Signer: %s is not permitted.",
+                         header.signer_pubkey)
+            return False
+
+        except AttributeError:
+            LOGGER.debug("Chain head not yet set")
+            return True
+
+    def is_transaction_signer_authorized(self, transactions, identity_view):
+        """ Check the transaction signing key against the allowed transactor
+            permissions. The roles being checked are the following, from first
+            to last:
+                "transactor.transaction_signer.<TP_Name>"
+                "transactor.transaction_signer"
+                "transactor"
+                "default"
+
+            The first role that is set will be the one used to enforce if the
+            transaction signer is allowed.
+
+            Args:
+                transactions (List of Transactions): The transactions that are
+                    being verified.
+                identity_view (IdentityView): The IdentityView that should be
+                    used to verify the transactions.
+        """
+        role = None
+        if role is None:
+            role = identity_view.get_role(
+                "transactor.transaction_signer")
+
+        if role is None:
+            role = identity_view.get_role("transactor")
+
+        if role is None:
+            policy_name = "default"
+        else:
+            policy_name = role.policy_name
+
+        policy = identity_view.get_policy(policy_name)
+
+        family_roles = {}
+        for transaction in transactions:
+            header = TransactionHeader()
+            header.ParseFromString(transaction.header)
+            family_policy = None
+            if header.family_name not in family_roles:
+                role = identity_view.get_role(
+                    "transactor.transaction_signer." + header.family_name)
+
+                if role is not None:
+                    family_policy = identity_view.get_policy(role.policy_name)
+                family_roles[header.family_name] = family_policy
+            else:
+                family_policy = family_roles[header.family_name]
+
+            if family_policy is not None:
+                if not self._allowed(header.signer_pubkey, family_policy):
+                    LOGGER.debug("Transaction Signer: %s is not permitted.",
+                                 header.signer_pubkey)
+                    return False
+            else:
+                if policy is not None:
+                    if not self._allowed(header.signer_pubkey, policy):
+                        LOGGER.debug(
+                            "Transaction Signer: %s is not permitted.",
+                            header.signer_pubkey)
+                        return False
+        return True
+
+    def _allowed(self, public_key, policy):
+        for entry in policy.entries:
+            if entry.type == Policy.PERMIT_KEY:
+                if public_key == entry.key or "*" in entry.key:
```

Comment:
`"*" in entry.key` will hold true for any string with a `*` in it.  Can  we specify partials? I.e. `"<partial key>*"`.  If that's the case, we need to do more of a match.  if not, we can probably just do an `==` here.